src/peft/utils/other.py

```diff
@@ -498,3 +498,23 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
         unique_id = storage_ptr(tensor)
 
     return tensor.device, unique_id, storage_size(tensor)
+
+
+def cast_non_trainable_to_dtype(model, dtype):
+    """
+    Cast all non-trainable parameters of the model to the given `dtype`. The trainable parameters are casted to full
```

Comment:
```suggestion
    Cast all non-trainable parameters of the model to the given `dtype`. The trainable parameters are cast to full
```

Reply:
Maybe worth mentioning that dtype should typically be bf16 or fp16? Are there other dtypes that make sense here?